src/main/java/dk/kb/util/yaml/YAML.java

```diff
@@ -651,20 +652,25 @@ public static YAML parse(InputStream yamlStream) {
      *
      * @param yamlPath path to YAML File.
      * @return a YAML based on the given stream.
+     * @throws IOException if a configuration could not be fetched.
+     * @throws FileNotFoundException if the given yamlFile could not be located.
      */
     public static YAML parse(Path yamlPath) throws IOException {
-        try (InputStream in = IOUtils.buffer(new FileInputStream(yamlPath.toFile()))) {
-            return parse(in);
-        }
+        return parse(yamlPath.toFile());
     }
     
     /**
      * Parse the given File as YAML.
      *
      * @param yamlFile path to YAML File.
      * @return a YAML based on the given stream.
+     * @throws IOException if a configuration could not be fetched.
+     * @throws FileNotFoundException if the given yamlFile could not be located.
      */
     public static YAML parse(File yamlFile) throws IOException {
+        if (!yamlFile.exists()) {
```

Comment:
While we are at it, should we consider the case where there file exists, but we are not allowed to read it?

Reply:
Yep. Added together with a Unit-test.